ApplicationCode/ProjectDataModel/Completions/RimFractureModel.cpp

```diff
@@ -507,6 +515,15 @@ void RimFractureModel::updateBarrierProperties()
     {
         clearBarrierAnnotation();
     }
+
+    if ( m_showOnlyBarrierFault() && m_hasBarrier() )
+    {
+        hideOtherFaults( m_barrierFaultName() );
+    }
+    else
+    {
+        showAllFaults();
```

Comment:
This is a bit problematic. If the user for some reason already has unselected some faults, this operation will reset this selection. Maybe add a toolbutton "Show All Faults" instead? Then this will be a direct command instead of state change.

**Suggestion**: Add a boolean field "Show All Fault", and use the concept for PdmUiToolButtonEditor caf::PdmUiToolButtonEditor::uiEditorTypeName() );

Reply:
Yes, was planning to discuss since it turned out a bit strange. Will try you suggestion.

Reply:
Fixed.